packages/openapi-generator/src/generator.ts

```diff
@@ -30,30 +32,34 @@ const logger = createLogger('openapi-generator');
  * Generates files using the OpenApi Generator CLI and wraps the resulting API in an SDK compatible API.
  * @param options Options to configure generation.
  */
-export async function generate(options: GeneratorOptions): Promise<void> {
+export async function generate(options: GeneratorOptions): Promise<void[]> {
   options.serviceMapping =
     options.serviceMapping ||
     resolve(options.input.toString(), 'service-mapping.json');
 
   if (options.clearOutputDir) {
-    await rmdir(options.outputDir, { recursive: true });
+    if (existsSync(options.outputDir)) {
```

Comment:
I am pretty sure this check is not necessary. The `recursive` option should already handle this.

Reply:
I just saw your funfact comment and crosschecked with [the documentation](https://nodejs.org/api/fs.html#fs_fs_rmdir_path_options_callback). According to this it should not break. Is it possible that there was another error?